lang/java/idl/src/main/java/org/apache/avro/idl/SchemaVisitorAction.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.avro.idl;
+
+public enum SchemaVisitorAction {
```

Comment:
if inspired from java.nio.file.FileVisitResult for walkFileTree(Path start, FileVisitor<? super Path> visitor), it could be renamed in SchemaVisitorResult ?

Reply:
Actually, it was copied from the `avro-compiler` package. So that's where the naming comes from.

It could be done differently (arguably better), but I chose to keep it as-is as much as possible.